### PR TITLE
Added doPrivileged calls to ThingManager and DiscoveryServiceRegistry.

### PR DESCRIPTION
**Pull request by dnobel**
Bug: 474692
Signed-off-by: Dennis Nobel d.nobel@external.telekom.de

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/internal/ThingLinkManagerOSGiTest.groovy [hunk 1]
```diff
@@ -95,50 +95,50 @@ class ThingLinkManagerOSGiTest extends OSGiTest{
...
         thingSetupManager.addThing(thingUID, new Configuration(), /* bridge */ null)
-        
+
         Thing thing = thingRegistry.get(thingUID)
         assertThat thing, is(notNullValue())
-        
-        thingSetupManager.removeThing(thingUID)
-        
+
+        thingSetupManager.removeThing(thingUID, true)
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/internal/ThingLinkManagerOSGiTest.groovy]: This has nothing to do with this PR, right?

File: bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/DiscoveryServiceRegistryImpl.java [hunk 2]
```diff
@@ -210,13 +212,19 @@ public synchronized void removeDiscoveryListener(DiscoveryListener listener) thr
...
             cachedResults.put(source, result);
         }
-        for (DiscoveryListener listener : this.listeners) {
+        for (final DiscoveryListener listener : this.listeners) {
             try {
-                listener.thingDiscovered(source, result);
+                AccessController.doPrivileged(new PrivilegedAction<Void>() {
+                    @Override
+                    public Void run() {
+                        listener.thingDiscovered(source, result);
```
Comment (kaikreuzer) [inline bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/DiscoveryServiceRegistryImpl.java]: I am not an expert on Java security, but doesn't calling all listeners in privilege mode mean that all you have to do is to register such a listener and you can run any malicious code inside it in privileged mode?

Comment (dnobel) [inline bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/DiscoveryServiceRegistryImpl.java]: What doPrivileged does is, that it executes the code inside the call (listener.thingDiscovered(source, result);) with the permission of the calling bundle (in that case the core.discovery bundle). If the implementation of the listener again has a security context this security context is applied.

File: bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/internal/ThingLinkManagerOSGiTest.groovy (hunk 1 above)
Comment (dnobel) [inline bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/internal/ThingLinkManagerOSGiTest.groovy]: No. that is the fixed test, which failed after I have added the doPrivileged call.

**Pull request merged**